### ISSUE TITLE
Missing keys in state_dict and testing results are all zero for compatible NTSCC+

### ISSUE THREAD

**Issue by asdLeaving** [wsxtyrdd/NTSCC_plus#4]
Missing keys when load state_dict for compatible NTSCC+:
`_IncompatibleKeys(missing_keys=['f_e0.blocks.1.attn_mask', 'f_e0.blocks.3.attn_mask', 'f_d0.blocks.1.attn_mask', 'f_d0.blocks.3.attn_mask', 'f_list.0.blocks.1.attn_mask', 'f_list.0.blocks.3.attn_mask', 'f_list.1.blocks.1.attn_mask', 'f_list.1.blocks.3.attn_mask', 'rate_adaption_enc.mask', 'ntc.sc_transform.mask', 'trans_sep_enc.blocks.1.attn_mask', 'trans_sep_enc.blocks.3.attn_mask', 'trans_sep_dec.blocks.1.attn_mask', 'trans_sep_dec.blocks.3.attn_mask'], unexpected_keys=[])`

Though Q_level is different in each testing result, the loss, PSNR and CBR are all zero. A part of the testing results is shown as follows:
```
2024-03-17 10:08:54;INFO;============= eta=0.15 =============
2024-03-17 10:08:54;INFO;----------------------------------
test_compatible_ntscc.py:45: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)
  values = np.linspace(math.log(max_val), math.log(min_val), num)
2024-03-17 10:08:54;INFO;Q_level 2.360935688018799 | Eta 0.15 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:54;INFO;Q_level 1.9882715034955487 | Eta 0.15 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:54;INFO;Q_level 1.6744308587794836 | Eta 0.15 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:54;INFO;Q_level 1.4101286951524608 | Eta 0.15 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:54;INFO;Q_level 1.1875455629991198 | Eta 0.15 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:54;INFO;Q_level 1.0000962813159555 | Eta 0.15 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.8422351133846508 | Eta 0.15 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.7092916946802956 | Eta 0.15 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.5973328588981317 | Eta 0.15 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.5030462741851807 | Eta 0.15 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 2.360935688018799 | Eta 0.19999999999999998 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.9882715034955487 | Eta 0.19999999999999998 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.6744308587794836 | Eta 0.19999999999999998 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.4101286951524608 | Eta 0.19999999999999998 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.1875455629991198 | Eta 0.19999999999999998 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.0000962813159555 | Eta 0.19999999999999998 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.8422351133846508 | Eta 0.19999999999999998 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.7092916946802956 | Eta 0.19999999999999998 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.5973328588981317 | Eta 0.19999999999999998 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.5030462741851807 | Eta 0.19999999999999998 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 2.360935688018799 | Eta 0.25 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.9882715034955487 | Eta 0.25 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.6744308587794836 | Eta 0.25 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.4101286951524608 | Eta 0.25 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.1875455629991198 | Eta 0.25 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.0000962813159555 | Eta 0.25 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.8422351133846508 | Eta 0.25 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.7092916946802956 | Eta 0.25 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.5973328588981317 | Eta 0.25 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.5030462741851807 | Eta 0.25 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 2.360935688018799 | Eta 0.3 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.9882715034955487 | Eta 0.3 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.6744308587794836 | Eta 0.3 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.4101286951524608 | Eta 0.3 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.1875455629991198 | Eta 0.3 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 1.0000962813159555 | Eta 0.3 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.8422351133846508 | Eta 0.3 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.7092916946802956 | Eta 0.3 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.5973328588981317 | Eta 0.3 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;Q_level 0.5030462741851807 | Eta 0.3 | SNR 0 | Loss 0.000 | PSNR 0.00 | CBR 0.0000 | CBR_y 0.0000
2024-03-17 10:08:55;INFO;======  SNR=0dB  =======
2024-03-17 10:08:55;INFO;PSNR: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
2024-03-17 10:08:55;INFO;CBR: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
```

**Comment by wsxtyrdd** [comment]
1. Missing keys: It's not an issue, as the attention mask will be recalculated based on the specific resolution of the input image.
2. Please verify your test dataset, as it appears that there are no images present in your specified test directory.

**Comment by asdLeaving** [comment]
> 1. Missing keys: It's not an issue, as the attention mask will be recalculated based on the specific resolution of the input image.
> 2. Please verify your test dataset, as it appears that there are no images present in your specified test directory.

Yes, the zero-value issue is caused by test directory error. The test directory needs to be specified in accordance with line 206, 207 and 211 of `test_compatible_ntscc.py`, which is different from the directory setting in `main.py`.
